test/core/test/each.test.ts

```diff
@@ -98,6 +98,28 @@ test.each([
   expect(result).toBe(expected)
 })
 
+test.each([
+  [1, 1, 2],
+  [1, 2, 3],
+  [2, 1, 3],
+])('the number of the test case is %$', (a, b, expected) => {
+  expect(a + b).toBe(expected)
+})
```

Comment:
Technically these tests only serve as examples and don't assert the feature. I'm not sure we have such tests for `each`, but we can probably add it to reporter like here

https://github.com/vitest-dev/vitest/blob/c82387d36f7dc0cf8f558b96bf97f782b0b3637a/test/reporters/tests/default.test.ts#L124-L133

Btw, I'm still a bit confused but the expected title output is this?

```
the number of the test case is 1
the number of the test case is 2
the number of the test case is 3
```